dipper/sources/ZFIN.py

```diff
@@ -2544,43 +2386,54 @@ def _load_zp_mappings(self, file):
         Given a file that defines the mapping between
         ZFIN-specific EQ definitions and the automatically derived ZP ids,
         create a mapping here.
-        This may be deprecated in the future
         :return:
+        dict zp_map with 
 
         """
         zp_map = {}
         LOG.info("Loading ZP-to-EQ mappings")
         line_counter = 0
         with open(file, 'r', encoding="utf-8") as csvfile:
             filereader = csv.reader(csvfile, delimiter='\t', quotechar='\"')
+            header_or_comment_re = re.compile('^iri|^#')
             for row in filereader:
+                if header_or_comment_re.match(row[0]):
+                    continue
                 line_counter += 1
-                (zp_id, zp_label, superterm1_id, subterm1_id, quality_id,
-                 modifier, superterm2_id, subterm2_id) = row
+
+                zp_id = row[0]
+                modifier = self.resolve("abnormal", False) # Nico assures us that id_map_zfin.tsv only contains data for abnormal phenotypes                
```

Comment:
This is true, but it is, in theory, possible that one day, normal phenotype statements are allowed. Lets be sure that:
1. We filter out everything that is not abnormal
2. We never make the assumption that anything we find in a ZFIN dataset IS in fact abnormal

Both must be true.

Reply:
Ah okay, I'll take this out - this was just a note I made to myself while we were meeting. It does seem a little hostile now that I read it! Sorry about that!

Reply:
Ah nooo! This was not what I read. :P Like you, I just want the ingest to be future-proof, thats all. :P

Reply:
we possibly then should make an item in your file explicitly stating "abnormal" or PATO:0000460 (even though every item will have it), and then pass out those that don't, to help prevent normal phenotypes from slipping into the ingest in the future...

Reply:
Hmmmmmmm... I think I have to admit you might be right. Can you do the following: when you parse the mapping file, after the split, if there is no 8th element in the split, assume it is PATO: 0000460, and when there is an 8th element, take that instead. That way we dont have to wait until I actually add the additional column, but if I do, and something other than PATO: 0000460 will be added, it will be robust and work.

Reply:
For now though: leave the abnormal filter in there. Just add an assumption that if there are only 7 elements in the split, the 8th is PATO: 0000460.

Reply:
we could do that, although it makes the code and zp-mapping-2019.txt a little more opaque and confusing for the next person who has to maintain the code...

Reply:
Actually I think it might also be a good idea to put a header in the zp-mapping file also explaining what each field is (also in the interest of making the code/data a bit more intuitive), but that's not a show stopper